src/main/java/org/snomed/snowstorm/extension/ExtensionAdditionalLanguageRefsetUpgradeService.java

```diff
@@ -154,67 +151,52 @@ private Map<Long, ReferenceSetMember> getReferencedComponents(String branchPath,
-		Map<Long, Set<Description>> conceptToDescriptionsMap = new Long2ObjectOpenHashMap<>();
+		descriptionIds.addAll(languageRefsetMembersToCopy.values().stream().map(item -> Long.valueOf(item.getReferencedComponentId())).collect(Collectors.toSet()));
+
+		// Get all existing language reference set members
+		Map<String, ReferenceSetMember> existingLanguageRefsetMembersToUpdate = new HashMap<>();
 		for (List<String> batch : partition(conceptIds, 10_000)) {
 			NativeSearchQueryBuilder queryBuilder = new NativeSearchQueryBuilder()
-					.withQuery(boolQuery().must(branchCriteria.getEntityBranchCriteria(Description.class)))
-					.withFilter(termsQuery(CONCEPT_ID, batch))
+					.withQuery(boolQuery()
+							.must(branchCriteria.getEntityBranchCriteria(ReferenceSetMember.class))
+							.must(termQuery(REFSET_ID, config.getDefaultEnglishLanguageRefsetId()))
+							.must(termsQuery(ACCEPTABILITY_ID_FIELD_PATH, Concepts.PREFERRED, Concepts.ACCEPTABLE))
+							.must(termsQuery(CONCEPT_ID, batch))
+					).withFields(REFERENCED_COMPONENT_ID, ACTIVE, ACCEPTABILITY_ID_FIELD_PATH)
 					.withPageable(LARGE_PAGE);
 
-			try (final SearchHitsIterator<Description> searchHitsIterator = elasticsearchTemplate.searchForStream(queryBuilder.build(), Description.class)) {
+			try (final SearchHitsIterator<ReferenceSetMember> searchHitsIterator = elasticsearchTemplate.searchForStream(queryBuilder.build(), ReferenceSetMember.class)) {
 				searchHitsIterator.forEachRemaining(hit -> {
-					descriptionIds.add(Long.valueOf(hit.getContent().getId()));
-					conceptToDescriptionsMap.computeIfAbsent(Long.valueOf(hit.getContent().getConceptId()), k -> new HashSet<>()).add(hit.getContent());
-					if (hit.getContent().getModuleId().equals(config.getDefaultModuleId())) {
-						extensionDescriptionIds.add(Long.valueOf(hit.getContent().getId()));
-					}
+					existingLanguageRefsetMembersToUpdate.put(hit.getContent().getReferencedComponentId(), hit.getContent());
+					descriptionIds.add(Long.valueOf(hit.getContent().getReferencedComponentId()));
 				});
 			}
 		}
 
-		// Get all existing language refset members
-		List<ReferenceSetMember> existingLanguageRefsetMembersToUpdate =  new ArrayList<>();
-		List<ReferenceSetMember> existingExtensionLanguageRefsetMembers =  new ArrayList<>();
+		// Get all relevant descriptions
+		Map<Long, Set<Description>> conceptToDescriptionsMap = new Long2ObjectOpenHashMap<>();
 		for (List<Long> batch : partition(descriptionIds, 10_000)) {
 			NativeSearchQueryBuilder queryBuilder = new NativeSearchQueryBuilder()
-					.withQuery(boolQuery()
-							.must(branchCriteria.getEntityBranchCriteria(ReferenceSetMember.class))
-							.must(termQuery(REFSET_ID, config.getDefaultEnglishLanguageRefsetId()))
-					).withFilter(termsQuery(REFERENCED_COMPONENT_ID, batch))
+					.withQuery(boolQuery().must(branchCriteria.getEntityBranchCriteria(Description.class)))
+					.withFilter(termsQuery(DESCRIPTION_ID, batch))
```

Comment:
@QuyenLy87 I don't think you need to fetch descriptions as description ids are sufficient.

Reply:
Because we need to know the description type @CoderMChu.  A preferred term could be an FSN, SYN, or DEF. So we have to check the preferred term in the same description type.

Reply:
@QuyenLy87 Yes that is a good point. In this case we can just run a query to check these are not FSN.